### DISCUSSION TITLE
Are you working on receiving messages also?

### DISCUSSION THREAD

**Discussion by ProgrammerNomad** [netflie/whatsapp-cloud-api#24] [Ideas]
Hello,

Thanks for making such great lib, i have tested that we can save reply also, so please make addon for receiving and saving chat and media files.

Thanks

**Comment by ProgrammerNomad** [comment]
i just want to parse incoming messages so i can make mu custom chatbot

**Reply by kvvaradha** [reply]
Receiving messages to be done through Webhooks. Did you configure your webhook and validated it with developer portal,?

**Reply by ProgrammerNomad** [reply]
Yes, i am Receiving messages but need a proper parser that could parse all type of messages and provide a array in return. There is some node js based lib they are providing Parser for Webhook received data.

**Reply by ProgrammerNomad** [reply]
pls see here they are proving: https://github.com/DaggieBlanqx/whatsappcloudapi_wrapper

**Comment by haroonakram** [comment]
about receiving messages and saving any update on this???

**Comment by aalbarca** [comment]
Hi,

I don't plan to create a wrapper, although I don't rule out the idea for the future. I'm sorry.

**Comment by aalbarca** [comment]
2.x version has been released! https://github.com/netflie/whatsapp-cloud-api/discussions/75 with webhook support @nomadicshiv
